### PR TITLE
[NMA-1019]  Firebase events for users and contacts

### PR DESCRIPTION
**Pull request by Syn-McJ**
## Issue being fixed or feature implemented
Added events for users and contacts and turned off logging for debug builds.

Additionally, I have moved some dashpay specific fragments and activities to view bindings, and replaced `TextWatcher` implementations with `doAfterTextChanged` extension method, which allows getting rid of the interface and unneeded methods implementations.

## Related PR's and Dependencies
https://github.com/dashevo/dash-wallet/pull/724
https://github.com/dashevo/dash-wallet/pull/737
https://github.com/dashevo/dash-wallet/pull/742
https://github.com/dashevo/dash-wallet/pull/753
https://github.com/dashevo/dash-wallet/pull/756

## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [x] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: wallet/src/de/schildbach/wallet/ui/dashpay/EditProfileViewModel.kt [hunk 1]
```diff
@@ -322,7 +322,6 @@ class EditProfileViewModel @Inject constructor(
                         config.imgurDeleteHash = imgurUploadResponse.data.deletehash
                         val avatarUrl = imgurUploadResponse.data.link
                         Log.d("AvatarUrl", avatarUrl)
-                        dashPayProfile?.avatarUrl = avatarUrl
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/ui/dashpay/EditProfileViewModel.kt:319]: This doesn't seem to be needed (it's not used in the similar `uploadToGoogleDrive` method and doesn't affect the logic), but it's messing up later checks on whether the new avatar URL is different from the original. Let me know if I'm missing something.

Comment (HashEngineering) [reply wallet/src/de/schildbach/wallet/ui/dashpay/EditProfileViewModel.kt:319]: Good question.  Let me see.

Comment (HashEngineering) [reply wallet/src/de/schildbach/wallet/ui/dashpay/EditProfileViewModel.kt:319]: I believe that you are correct.  `dashPayProfile` contains the current profile data from the Room database and we shouldn't be changing any of its member fields.  It should only updated when the room database is updated.

If we change the image, then that image url is stored here `profilePictureUploadLiveData.value!!.data` after uploading.  When "Save" is pressed, then the updated profile with the new images or text is broadcast to the Platform Network and it is also saved to the Room database.

**Review by HashEngineering — APPROVED** [review]
Excellent.

**Pull request merged**